app/src/main/java/io/neurolab/PlaybackStream.java

```diff
@@ -0,0 +1,129 @@
+package io.neurolab;
+
+import android.content.Context;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import io.neurolab.tools.ResourceManager;
+
+public class PlaybackStream implements InputInterface {
+
+    static DataReceiver receiver;
+    static int currentIndex = 0;
+    private int numberOfChannels;
+    private boolean loopPlayback = false;
+    private Timer timer;
+    static Random random = new Random();
+    static ArrayList<ArrayList<double[]>> data;
+
+    private Context context;
+
+    public PlaybackStream(Context context, DataReceiver receiver, int numberOfChannels, String playbackFile, boolean loop) {
+        this.context = context;
+        this.receiver = receiver;
+        this.numberOfChannels = numberOfChannels;
+        this.setPlaybackFile(playbackFile, loop);
+
+    }
+
+    @Override
+    public boolean sendCommand(String string) {
+        return true;
+    }
+
+    public void setPlaybackFile(String file, boolean loop) {
+        System.out.println("file:" + file);
+        File playbackFile = ResourceManager.getInstance().getResource(context, file);
+
+        this.loopPlayback = loop;
+        this.data = new ArrayList<>();
+
+        if (!playbackFile.exists()) {
+            for (int i = 0; i < 256; i++) {
+                System.out.println("i:" + i);
+                ArrayList<double[]> currentSamples = new ArrayList<>();
+                double[] currentSample = new double[4];
+                for (int c = 0; c < this.numberOfChannels; c++) {
+                    currentSample[c] = 12.5d * Math.sin(((double) (i + c * 16) / 128d) * Math.PI * 6d);
+                    currentSample[c] += 12.5d * Math.sin(((double) (i + c * 16) / 128d) * Math.PI * 25d);
+                }
+                currentSamples.add(currentSample.clone());
+                this.data.add(currentSamples);
+            }
+        } else {
+            try {
```

Comment:
Use try with resources

Reply:
Can you please elaborate more?

Reply:
Read about try with resources

Reply:
Okay thanks, I will check it out and implement it 👍